### PR TITLE
Wait results option

### PR DESCRIPTION
**Pull request by xBeAsTx**
So, sometime it's usefull send task and get results in same http-query. Moreover, it allows not to call every 1-2-3 seconds for requesting results by '/api/task/result/' method.

This pull request implements it.

Example:
```
curl -X POST -d '{"args":["..."]}' http://127.0.0.1:5555/api/task/apply/task.foo
```

### CODE REVIEW

File: flower/api/tasks.py [hunk 1]
```diff
@@ -113,6 +119,10 @@ def post(self, taskname):
 
         result = task.apply_async(args=args, kwargs=kwargs, **options)
         response = {'task-id': result.task_id}
+        if 'wait' in options:
+            # wait until task finished and do not raise anything
+            result.get(propagate=False)
```
Comment (mher) [inline flower/api/tasks.py]: result.get will block the event loop and other requests couldn't be handled

**Comment by xBeAsTx** [comment]
Now it's asynchronously

File: flower/api/tasks.py [hunk 2]
```diff
@@ -113,9 +120,25 @@ def post(self, taskname):
...
+
+    def async_wait(self, result, response):
+        from threading import Thread
+        def wait_results(result, response):
+            # wait until task finished and do not raise anything
+            r = result.get(propagate=False)
+            self.update_response_result(response, result)
+            self.write_result(result, response)
+        th = Thread(target=wait_results, args=(result, response, ))
+        th.start()
```
Comment (mher) [inline flower/api/tasks.py]: th.join() is missing here

**Comment by mher** [comment]
Can you move this functionality into a separate handler and document it? /api/task/apply/(.+) seems a good candidate.

**Comment by mher** [comment]
Merged, thanks!

**Pull request merged**